docs/core/PINs/PIN-16-Obfuscating-Data-Passed-to-Cloud.md

```diff
@@ -0,0 +1,161 @@
+
+```
+    x logger.info("this is sensitive")
+    |
++---+
+|   |
+|   |
+|   logged: "INFO: this is sensitive"
+|
+logged: "INFO: this is ****"
+```
+
+While developing this PIN, a fifth scenario was postponed to a later PIN. All considered API designs for (5) were tightly coupled to a more complex abstraction that allowed information from handlers to be passed to other handlers. Due to its complexity, we will propose options for (5) in a secondary PIN for independent debate.
+
+5. **Value Relocation**: fully redact content from the LogRecord and annotate where the user can find where sensitive information has been logged to. 
+
+```
+    x logger.info("this is sensitive", private=True)
+    |
++---+
+|   |
+|   |
+|   logged: "INFO: please see 'bucket-name/path/file' for more info"
+|
+logged @ 'bucket-name/path/file': "INFO: this is sensitive"
+```
+
+In addition, two scenarios were specifically dropped from consideration. (6) and (7) protect information by filtering LogRecords entirely from a handler’s emit method. Since this proposal is inspired by users who _do_ want to use Cloud logging, just on their own terms, these scenarios are considered irrelevant.
+
+6. **Simple Filtering**: all or nothing inclusion or removal of the LogRecord. 
+
+```
+   logger.info("this is sensitive", private=True)
+   |
+   |
+   X (not logged)
+```
+
+7. **Per-handler Filtering**: all or nothing inclusion or removal of LogRecords for specific Handlers. 
+```
+    x logger.info("this is sensitive")
+    |
++---+
+|   |
+|   |
+|   logged: "this is sensitive"
+|
+X (not logged)
+```
+
+## Proposal
+
+::: tip
+During the draft phase, these proposals are purposefully listed in outline form to enable precise Github comments.
+:::
+
+### Design goals
+- Obfuscation behavior driven by prefect configuration: A global setting can obfuscate all task logs and exception traces from Cloud.
+- Obfuscation behavior customizable at many levels: Redaction patterns, obfuscation functions, and setting privacy at log levels or per handler allow the user to customize obfuscation behavior.
+- Warn users as early as possible if obfuscation mechanisms have been elected but there could be a known potential leak of information.
+- Utilize as much of the Python standard logging subsystem as possible
+- Don’t break the existing Cloud UI: The error log tile functionality in Cloud UI is not broken; it can still report number of errors and link to obfuscated error logs in Cloud UI like it does today
+
+### Primary proposals:
+#### Obfuscate Logs sent to Prefect Cloud
+- User can elect all logs to be obfuscated in entirety, for example by setting `prefect.config.logger.set_all_private`. This would still send a LogRecord to Cloud indicating the log level, however, all content from a record’s `msg` attribute would be removed. This is useful for the Cloud UI to determine if there is a log entry indicating a failure within a Flow Run.
```

Comment:
Could we consider using a variable with a fixed number of settings rather than a boolean? I'm imagining `prefect.config.logger.redaction_level = {'NONE', 'PRIVATE', 'ALL', 'PARTIAL', 'SOME_OTHER_LEVEL'}` etc. in order to allow this one config to represent possibly multiple default levels.